### PR TITLE
[Datasource] Add Hostgroup

### PR DESCRIPTION
**Pull request by michaelkad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPiHostgroupDataSourceBasic
--- PASS: TestAccIBMPiHostgroupDataSourceBasic (18.45s)
PASS

...
```

### CODE REVIEW

File: ibm/service/power/ibm_pi_constants.go [hunk 1]
```diff
@@ -29,8 +29,10 @@ const (
 	Arg_VolumeGroupID                       = "pi_volume_group_id"
 	Arg_VolumeID                            = "pi_volume_id"
 	Arg_VolumeOnboardingID                  = "pi_volume_onboarding_id"
+	Arg_HostgroupID                         = "pi_hostgroup_id"
 
 	// Attributes
+	Attr_Hosts                                       = "hosts"
```
Comment (ismirlia) [inline ibm/service/power/ibm_pi_constants.go:35]: Can you resort the new constants with the rest of the constants?

Comment (michaelkad) [reply ibm/service/power/ibm_pi_constants.go]: Done

File: website/docs/d/pi_host_group.html.markdown [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- `id` - The unique identifier of the host group.
+- `creation_date` - (String) Date/Time of host group creation.
+
+- `hosts` - (List) List of hosts.
+
+- `name` - (String) Name of the host group.
+
+- `primary` - (String) Name of the workspace owning the host group.
+
+- `secondaries` - (List) Names of workspaces the host group has been shared with.
```
Comment (ismirlia) [inline website/docs/d/pi_host_group.html.markdown:59]: I think this description has been updated say workspace IDs in the model.

File: ibm/service/power/data_source_ibm_pi_host_group.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_HostgroupID: {
```
Comment (ismirlia) [inline ibm/service/power/data_source_ibm_pi_host_group.go:30]: Style needs to be fixed, should be "Arg_HostGroupID"

**Comment by michaelkad** [comment]
combine with #87

**Pull request closed without merging**